CMakeLists.txt

```diff
@@ -2,6 +2,17 @@ cmake_minimum_required(VERSION 3.14)
 
 #set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CMAKE_COMMAND} -E time")
 
+# determine whether it's main/root project
+# or being built under another project.
+if (NOT DEFINED BOOST_REDIS_MAIN_PROJECT)
```

Comment:
This check will be always evaluated to false? Why can't we simply write

```
   if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
      set(BOOST_REDIS_MAIN_PROJECT ON)
   else()
      set(BOOST_REDIS_MAIN_PROJECT OFF)
```

Reply:
In case for some reason user needed to define it manually, it would skip reassigning. specially when not using `add_subdirectory` but requiring the same behavior as if it was a subproject, for example using the library from another build system.

I personally haven't came across such situations, and I've just copied and edited it for this project. but my guess is the above.